File: bench/ref/mandel.go
package main
import "fmt"
func mandel(size int) int64 { var total int64=0
  for py:=0;py<size;py++ { for px:=0;px<size;px++ {
    x0:=float64(px)/float64(size)*3.5-2.5; y0:=float64(py)/float64(size)*2.0-1.0
    x,y:=0.0,0.0; it:=int64(0)
    for it<1000 && x*x+y*y<=4.0 { xt:=x*x-y*y+x0; y=2*x*y+y0; x=xt; it++ }
    total+=it } }; return total }
func main(){ fmt.Println(mandel(800)) }

File: bench/ref/mandel.py
def mandel(size):
    total=0
    for py in range(size):
        for px in range(size):
            x0=px/size*3.5-2.5; y0=py/size*2.0-1.0; x=0.0; y=0.0; it=0
            while it<1000 and x*x+y*y<=4.0:
                x,y=x*x-y*y+x0, 2*x*y+y0; it+=1
            total+=it
    return total
print(mandel(800))
